crates/sargon/src/system/sargon_os/sargon_os_accounts.rs

```diff
@@ -331,34 +331,56 @@ impl SargonOS {
         &self,
         account_address: AccountAddress,
     ) -> Result<()> {
-        let mut account = self.account_by_address(account_address)?;
-        account.mark_as_hidden();
-        self.update_account(account).await
+        self.update_profile_with(|profile| {
+            profile.networks.hide_account(&account_address);
+            Ok(())
+        })
+        .await?;
+
+        self.clients
+            .profile_state_change
+            .emit(ProfileState::Loaded(self.profile()?))
```

Comment:
😞 at least put a `// TODO: This is hacky and temporary and should be removed in the future, replaced by finer granularity Events` or similar?

Reply:
Right, but still ought to be a function, with good name; and we document it, why we need it.

Reply:
Move to `update_profile_with`. You mean to add the reasoning outlined in the ticket -> https://radixdlt.atlassian.net/browse/ABW-3591?

Reply:
Or am I missing something?

Reply:
> the emit call is not needed at all.

Not sure what you mean, the hosts need be notified with the profile is updated internally by Sargon. Up until this PR the hosts where notified only on wallet creation, and when the Hosts called `set_profile` after doing the mutation on their side.

Reply:
duh 🤦‍♂️ haha I missed we did not call inside of 

```rust
     match state {
            ProfileState::Loaded(ref mut profile) => mutate(profile),
            _ => Err(CommonError::ProfileStateNotLoaded {
                current_state: state.to_string(),
            }),
        }
```

ok so my point is, instead of "sprinkling" emits, why dont we add a single emit inside of that match? inside of `update_profile_with` in `ProfileStateHolder` ?

Reply:
I mentioned above that I moved it in [sargon_os update_profile_with](https://github.com/radixdlt/sargon/blob/b4fbe078d80867c34f561035001e34e7dce1652a/crates/sargon/src/system/sargon_os/sargon_os_profile.rs#L145).